samples/snippets/test_certificate_authorities.py

```diff
@@ -80,8 +81,10 @@ def test_delete_certificate_authority(capsys: typing.Any) -> None:
         PROJECT, LOCATION, CA_POOL_NAME, CA_NAME, COMMON_NAME, ORGANIZATION, CA_DURATION
     )
     delete_certificate_authority(PROJECT, LOCATION, CA_POOL_NAME, CA_NAME)
+    undelete_certificate_authority(PROJECT, LOCATION, CA_POOL_NAME, CA_NAME)
+    delete_certificate_authority(PROJECT, LOCATION, CA_POOL_NAME, CA_NAME)
     delete_ca_pool(PROJECT, LOCATION, CA_POOL_NAME)
```

Comment:
Consider moving these to to a pytest fixture that:

- creates the CA
- does the first delete
- yields the name
- (after test) - tries to delete again just in case and deletes ca pool and/or any other resources the test needs

Reply:
Have updated the test (new fixture: deleted_certificate_authority)